src/prefect/task_engine.py

```diff
@@ -249,6 +249,16 @@ def begin_run(self):
         new_state = Running()
         state = self.set_state(new_state)
 
+        # TODO: this is temporary until the API stops rejecting state transitions
+        # and the client / transaction store becomes the source of truth
+        # this is a bandaid caused by the API storing a Completed state with a bad
+        # result reference that no longer exists
+        if state.is_completed():
+            try:
+                state.result(retry_result_failure=False, _sync=True)
```

Comment:
I added the `retry_result_failure` passthrough kwarg to avoid retrying latency in this particular situation